src/tests/LitePsmJob-integration.t.sol

```diff
@@ -0,0 +1,166 @@
+    function transfer(address, uint256) external;
+    function transferFrom(address, address, uint256) external;
+}
+
+interface VatLike {
+    function frob(bytes32, address, address, address, int256, int256) external;
+    function hope(address) external;
+    function slip(bytes32, address, int256) external;
+    function ilks(bytes32) external view returns (uint256, uint256, uint256, uint256, uint256);
+    function debt() external view returns (uint256);
+    function Line() external view returns (uint256);
+    function urns(bytes32, address) external view returns (uint256, uint256);
+    function file(bytes32 ilk, bytes32 what, uint data) external;
+    function file(bytes32 what, uint data) external;
+}
+
+contract LitePsmJobIntegrationTest is DssCronBaseTest {
+
+    using GodMode for *;
+
+    uint256 constant MILLION_WAD = MILLION * WAD;
+
+    LitePsmLike public litePsm;
+    LitePsmJob public litePsmJob;
+    address public gem;
+    address public dai;
+    address public pocket;
+    address vat;
+    bytes32 ilk;
+
+    // --- Events ---
+    event Chug(uint256 wad);
+    event Fill(uint256 wad);
+    event Trim(uint256 wad);
+    event Work(bytes32 indexed network);
+
+    function setUpSub() virtual override internal {
+        litePsm = LitePsmLike(dss.chainlog.getAddress("MCD_LITE_PSM_USDC_A"));
+        pocket = dss.chainlog.getAddress("MCD_LITE_PSM_POCKET_USDC_A");
+        dai = dss.chainlog.getAddress("MCD_DAI");
+        litePsmJob = new LitePsmJob(address(sequencer), litePsm, MILLION_WAD, MILLION_WAD, MILLION_WAD);
+        gem = litePsm.gem();
+        ilk = litePsm.ilk();
+        vat = litePsm.vat();
+        // give auth access to this contract (caller) to vat for manipulating params
+        GodMode.setWard(vat, address(this), 1);
+    }
+
+    function test_fill() public {
+        // GodMode.setWard(address(litePsm), address(this), 1);
+        (uint256 Art,,, uint256 line,) = VatLike(vat).ilks(ilk);
+
+        // tArt must be greater than Art
+        // tArt = GemLike(gem).balanceOf(pocket) * GemConversionFactor + buf_;
+        uint256 GemConversionFactor = 10 ** (18 - GemLike(gem).decimals());
+        deal(gem, pocket, Art * 2 / GemConversionFactor);
+
+        // ilk line must be greater than Art
+        uint256 newLine = Art * 2 * RAY;
+        VatLike(vat).file(ilk, "line", newLine);
+        (Art,,, line,) = VatLike(vat).ilks(ilk);
+
+        // vat.Line() must be greater than vat.debt()
+        uint256 vatLine = VatLike(vat).Line();
+        uint256 vatDebt = VatLike(vat).debt();
+        VatLike(vat).file("Line", vatLine + vatDebt);
+
+        uint256 wad = litePsm.rush();
+        assertTrue(wad != 0);
+        (bool canWork, bytes memory args) = litePsmJob.workable(NET_A);
+        if (canWork){
+            (bytes4 fn) = abi.decode(args, (bytes4));
+            bytes4 encodedSelector = bytes4(abi.encode(litePsm.fill.selector));
+            assertEq(fn, encodedSelector);
```

Comment:
This could be rewritten simply as:
```suggestion
            (bytes4 fn) = abi.decode(args, (bytes4));
            assertEq(fn, litePsm.fill.selector);
```

Reply:
good catch! addressed in [f05df30](https://github.com/makerdao/dss-cron/pull/25/commits/f05df30d9f7ea524df6d252cdcd68dc6767cead0)